components/molecule/dataCounter/src/index.js

```diff
@@ -0,0 +1,196 @@
+import React, {useState} from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+import AtomButton from '@schibstedspain/sui-atom-button'
+import AtomInput, {inputSizes} from '@s-ui/react-atom-input'
+import MoleculeField from '@s-ui/react-molecule-field'
+
+const BUTTON_TYPE = 'secondary'
+
+const BASE_CLASS = `sui-MoleculeDataCounter`
+const CLASS_INPUT_CONTAINER = `${BASE_CLASS}-container`
+
+const isCharDigit = char => /[0-9]/.test(char)
+
+const MoleculeDataCounter = ({
+  id,
+  label,
+  value,
+  errorText: errorTextProps,
+  size = inputSizes.MEDIUM,
+  charsSize = 2,
+  max = 99,
+  min = 1,
+  minValueHelpText,
+  minValueErrorText,
+  maxValueHelpText,
+  maxValueErrorText,
+  onChange,
+  disabled
+}) => {
+  if (value) value = String(value)
+  else if (min) value = String(min)
+  else value = '0'
+
+  const [internalValue, setInternalValue] = useState(value)
+
+  const numInternalValue = Number(internalValue)
+  const numMax = Number(max)
+  const numMin = Number(min)
+
+  const isBelowMaxValue = numInternalValue < numMax
+  const isOverMinValue = numInternalValue > numMin
+  const isMaxValue = numInternalValue === numMax
+  const isMinValue = numInternalValue === numMin
+  const isLowerThanMinValue = numInternalValue < numMin
+  const isHigherThanMaxValue = numInternalValue > numMax
+
+  const decrementDisabled = disabled || numInternalValue <= numMin
+  const incrementDisabled = disabled || numInternalValue >= numMax
+
+  const incrementValue = e => {
+    if (isBelowMaxValue) {
+      const newValue = internalValue === '' ? min : parseInt(internalValue) + 1
+      const strNewValue = String(newValue)
+      setInternalValue(strNewValue)
+      onChange(e, {value: strNewValue})
+    }
+  }
+
+  const decrementValue = e => {
+    if (isOverMinValue) {
+      const newValue = internalValue === '' ? min : parseInt(internalValue) - 1
+      const strNewValue = String(newValue)
+      setInternalValue(strNewValue)
+      onChange(e, {value: strNewValue})
+    }
+  }
+
+  const removeDigit = e => {
+    if (internalValue.length) {
+      const newValue = internalValue.slice(0, -1)
+      setInternalValue(newValue)
+      onChange(e, {value: newValue})
+    }
+  }
+
+  const addDigit = (e, {key: digit}) => {
+    const newValue = internalValue + digit
+    if (newValue.length <= 2) {
+      setInternalValue(newValue)
+      onChange(e, {value: newValue})
+    }
+  }
+
+  const handleKeyDown = e => {
```

Comment:
Why not using {value} coming from AtomInput?
I can't add a digit before other. It always is being added at the end

Reply:
Right! I have improved both the code and the UX based on your comments